android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ImageProcessorFragment.java

```diff
@@ -1338,10 +1365,19 @@ private boolean getAppInstList() throws InterruptedException, ExecutionException
                 //Find fqdnPrefix from Port structure.
                 String fqdnPrefix = "";
                 List<Appcommon.AppPort> ports = appInst.getPortsList();
-                for (Appcommon.AppPort aPort : ports) {
+                // Get data only from first port.
+                Appcommon.AppPort aPort = ports.get(0);
```

Comment:
There's only one port (and the first one?)

Reply:
There should be 2 ports in the ComputerVision app definition: 8008 for REST and WS. 8009 for Persistent_Socket, my made-up protocol which probably should go away at some point. We only care about the first port for determining the prefix and whether we are TLS.